datastore/badger/config.go

```diff
@@ -81,6 +112,12 @@ func (cfg *Config) LoadJSON(raw []byte) error {
 
 func (cfg *Config) applyJSONConfig(jcfg *jsonConfig) error {
 	config.SetIfNotDefault(jcfg.Folder, &cfg.Folder)
+	if v, ok := correctLoadingMode(jcfg.TableLoadingMode); ok {
+		cfg.BadgerOptions.TableLoadingMode = options.FileLoadingMode(v)
```

Comment:
casting `options.FileLoadingMode` twice. Also, shouldn't it error ?

Reply:
You can make `jsonConfig.TableLoadingMode` a pointer to decide whether it's set at all (with a possible wrong value) or not.